src/v2/Apps/Settings/Routes/EditSettings/SettingsEditSettingsInformation.tsx

```diff
@@ -25,149 +25,115 @@ export const SettingsEditSettingsInformation: React.FC<SettingsEditSettingsInfor
   const { sendToast } = useToasts()
   const { submitUpdateSettingsInformation } = useUpdateSettingsInformation()
 
-  const onSubmit = async (
-    { email, name, phone, password },
-    // TODO: Fix typing & type errors
-    formikBag: FormikProps<any>
-  ) => {
-    formikBag.setStatus({ error: undefined })
-
-    try {
-      const variables = {
-        email,
-        name,
-        password,
-        phone,
-      }
-
-      const response = await submitUpdateSettingsInformation(variables)
-
-      const userOrError = response.updateMyUserProfile!.userOrError
-
-      sendToast({
-        variant: "success",
-        message: "Information updated successfully",
-      })
-
-      // TODO: Simplify this mess
-      if (userOrError!.mutationError) {
-        const { message, fieldErrors } = userOrError!.mutationError
-        if (fieldErrors) {
-          // display errors for a specified form field
-          const formattedErrors = formatGravityErrors(
-            userOrError!.mutationError
-          )
-          formikBag.setErrors(formattedErrors)
-        } else if (message) {
-          // display generic gravity error
-          formikBag.setStatus({ error: message })
-        }
-      } else {
-        formikBag.resetForm()
-      }
-    } catch (err) {
-      formikBag.setErrors(err)
-
-      sendToast({
-        variant: "error",
-        message: "There was a problem",
-        description: err.message,
-      })
-    }
-  }
-
   return (
     <>
       <Text variant="lg" mb={4}>
         Information
       </Text>
 
       <Formik
+        validateOnBlur
+        validateOnChange
         initialValues={{
-          name: me.name,
-          email: me.email,
-          phone: me.phone,
-          paddleNumber: me.paddleNumber,
-          internalID: me.internalID,
+          name: me.name ?? "",
+          email: me.email ?? "",
+          phone: me.phone ?? "",
+          password: "",
+        }}
+        validationSchema={Yup.object().shape({
+          email,
+          name,
+          // Requires password when email is changed
+          password: password.when("email", {
+            is: email => email !== me.email,
+            otherwise: field => field.notRequired(),
+          }),
```

Comment:
I added this conditional password validation — if the email isn't the email initially returned from the server, then the password is required. If _somehow_ a user is able to submit the form before the password field shows itself, this will prevent submission.